### PR TITLE
Migrate to latest Rust dependencies

### PR DESCRIPTION
**Pull request by str4d**
This brings in the Orchard dependencies, as well as a `metrics` feature we upstreamed.

### CODE REVIEW

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
Not sure why zkbot returned a failure; the linked CI run was successful.

File: Cargo.toml [hunk 1]
```diff
@@ -20,34 +20,31 @@ path = "src/rust/src/rustzcash.rs"
...
 hyper = { version = "=0.14.2", default-features = false, features = ["server", "tcp", "http1"] }
 ipnet = "2"
-metrics = "0.14.2"
-metrics-exporter-prometheus = "0.3"
+metrics = "0.16"
+metrics-exporter-prometheus = "0.5"
 thiserror = "1"
 tokio = { version = "1.0", features = ["rt", "net", "time", "macros"] }
 
-# Temporary workaround for https://github.com/myrrlyn/funty/issues/3
```
Comment (nuttycom) [inline Cargo.toml:48]: :tada:

**Review by nuttycom — APPROVED** [review]
utACK

**Review by ebfull — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -57,3 +54,9 @@ features = ["ansi", "chrono", "env-filter"]
 lto = true
 panic = 'abort'
 codegen-units = 1
+
+[patch.crates-io]
+ed25519-zebra = { git = "https://github.com/ZcashFoundation/ed25519-zebra.git", rev = "d3512400227a362d08367088ffaa9bd4142a69c7" }
+orchard = { git = "https://github.com/zcash/orchard.git", rev = "cd1e72bbcd9f2e873408aa365537c89824cb7430" }
```
Comment (str4d) [inline Cargo.toml:60]: This patch exists in order to get the dependency tree pinned correctly. We don't actually depend on the `orchard` crate yet, hence no change to `contrib/debian/copyright` re: BOSL.

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
Lesson learned: zkbot decides what branch it will merge to when the PR is opened, and doesn't update this when the GitHub PR is updated.

**Pull request merged**